tests/utils/fixtures/fixtures.py

```diff
@@ -155,14 +157,18 @@ def qemu_finalizer_impl(conn):
                 conn.run("poweroff")
                 halt_time = time.time()
                 # Wait up to 30 seconds for shutdown.
-                while halt_time + 30 > time.time() and qemu.poll() is None:
+                while (
+                    halt_time + shutdown_timeout_s > time.time() and qemu.poll() is None
+                ):
                     time.sleep(1)
             except:
                 pass
 
             # Terminate qemu
             try:
-                qemu.terminate()
+                os.killpg(os.getpgid(qemu.pid), signal.SIGTERM)
```

Comment:
I had to check, and this should actually work, since we [pass the `start new session` flag](https://github.com/mendersoftware/mender-image-tests/blob/master/tests/utils/common/common.py#L194) when we create the process. Funny that we already added it, but we didn't think about killing the process group here!